microservices/Keli-Katheya-User-Registration/src/main/java/com/kelikatheya/microservices/userRegistration/services/impls/UserServiceImpl.java

```diff
@@ -28,21 +32,44 @@
     @Autowired
     UserRepository userRepository;
 
+    @Autowired
+    private ConfirmationTokenRepository confirmationTokenRepository;
+
+    @Autowired
+    private EmailSenderService emailSenderService;
+
     @Autowired
     private MongoTemplate mongoTemplate;
 
     @Override
     public UserModel registerUser(UserRequest userRequest) {
+            UserEntity existingUser = userRepository.findByEmail(userRequest.getEmail());
+            if(existingUser!=null){
+                System.out.println("Already existing User");
```

Comment:
Its better we throw an exception and handle it to return error response. So that client will get that info